Aspen-Discovery/aspen-discovery
# 25.01.00 libkey integration for ebsco records

Jira ticket: [DIS-138](https://aspen-discovery.atlassian.net/browse/DIS-138)
---------------

Setting up LibKey before testing:

- navigate to Aspen Administration > Third Party
Enrichment > LibKey Settings
- create a new Setting (note: you will need
LibKey test credentials - can be requested from
ThirdIron)
- insert your credentials and add the library/ies
you plan to test the rest of the integration with
- save

For EBSCO host:
set up: see [https://drive.google.com/file/d/1PjoYk4pxaeuUxVLmHc3mFG9YEg1YG7g6/view](https://drive.google.com/file/d/1PjoYk4pxaeuUxVLmHc3mFG9YEg1YG7g6/view)

For EBSCO EDS
set up: see [https://aspen.bywatersolutions.com/education/setting-up-ebsco-eds](https://aspen.bywatersolutions.com/education/setting-up-ebsco-eds)

Test plan:

before applying the patch: 
- run a search in 'Article and Databases'
- take note of the covers images
- take note of the buttons available on each record

after applying the patch:
- notice the covers have now changed
- notice the new 'Access Online' link to the right of 'More Info' on each result
- check that the link opens the selected resource in a new tab (this should take you through LibKey and straight to the PDF document (if a pdf is available for the resource)
- find a retracted article, notice that a 'Retracted' warning message appears, that it is a link leading to LibKey page with information on the retraction, and that the 'Access Online' button is not available. 

 (if there are no retracted articles in your search results, you may want to temporarily set the doi in LibKeyDriver on line 13 to the doi of an article which you know is retracted, I used [10.1126/science.1124755](https://doi.org/10.1126/science.1124755). Note that this will apply to every single record in your search.)



[DIS-138]: https://aspen-discovery.atlassian.net/browse/DIS-138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
